### ISSUE TITLE
Executing "CATALOG_BRANCH="main" make clean cli" or "CATALOG_BRANCH=<feature-branch> make clean cli" leading to mulitple errors

### ISSUE THREAD

**Issue by prasad-shirodkar** [oracle-cne/ocne#216]
When executing command `CATALOG_BRANCH="main" make clean cli` or `CATALOG_BRANCH=<feature-branch> make clean cli`, below observations are seen:
The repo https://github.com/oracle-cne/catalog.git is being cloned twice as branches `main` and `developer`
```
ocne(main) $ CATALOG_BRANCH="main" make clean cli
rm -rf pkg/catalog/embedded/charts
rm -rf build/charts
rm -rf build
rm -rf out
mkdir -p build
git clone -b main  https://github.com/oracle-cne/catalog.git build/catalog
Cloning into 'build/catalog'...
remote: Enumerating objects: 2553, done.
remote: Counting objects: 100% (413/413), done.
remote: Compressing objects: 100% (240/240), done.
remote: Total 2553 (delta 222), reused 320 (delta 165), pack-reused 2140 (from 1)
Receiving objects: 100% (2553/2553), 2.43 MiB | 34.11 MiB/s, done.
Resolving deltas: 100% (1518/1518), done.
mkdir -p build
git clone -b developer  https://github.com/oracle-cne/catalog.git build/developer-catalog
Cloning into 'build/developer-catalog'...
remote: Enumerating objects: 2553, done.
remote: Counting objects: 100% (413/413), done.
remote: Compressing objects: 100% (240/240), done.
remote: Total 2553 (delta 222), reused 320 (delta 165), pack-reused 2140 (from 1)
Receiving objects: 100% (2553/2553), 2.43 MiB | 36.08 MiB/s, done.
Resolving deltas: 100% (1518/1518), done.
cp -r build/developer-catalog/charts/* build/catalog/charts
cp build/developer-catalog/olm/icons/* build/catalog/olm/icons
cd build/catalog && make
make[1]: Entering directory '/home/opc/projects/work/oracle-cne/ocne/build/catalog'
....
....
....
```

Note: The make target fails with error regarding olvm-capi chart.
```
[ "olvm-capi-1.0.0" = "$(yq '.name' charts/olvm-capi-1.0.0/Chart.yaml)-$(yq '.version' charts/olvm-capi-1.0.0/Chart.yaml)" ]
[ "$(yq '.version' charts/olvm-capi-1.0.0/Chart.yaml)" = "$(yq '.appVersion' charts/olvm-capi-1.0.0/Chart.yaml)" ]
grep "$(yq '.name' charts/olvm-capi-1.0.0/Chart.yaml)[[:space:]]|[[:space:]].*$(yq '.version' charts/olvm-capi-1.0.0/Chart.yaml).*|$" README.md
make[1]: *** [Makefile:17: repo/./olvm-capi-1.0.0.tgz] Error 1
make[1]: Leaving directory '/home/opc/projects/work/oracle-cne/ocne/build/catalog'
make: *** [Makefile:103: build/catalog/repo] Error 2
```

Relates to #208

**Comment by prasad-shirodkar** [comment]
It looks like `DEVELOPER_BUILD=false` needs to be set when building a non-release branch like main or feature branch....i.e. any branch not having prefix `release/`
`CATALOG_BRANCH="main" DEVELOPER_BUILD=false make clean cli`

**Comment by mgianatagh** [comment]
Cloning the repo twice is expected.  The `main` and `developer` branches are combined by default for non-release builds.  There is no single branch that contains the combined content of those two branches.

**Comment by mgianatagh** [comment]
The failure is due to some validation checks that were recently added to the `main` branch of catalog, but have not been merged to the release/2.0 branch.  It will fail on any branch of the catalog that has the new validation checks.

The snippet below is comparing chart versions found on the CATALOG_BRANCH, with a support matrix that was added to README.md.  For a developer build, the charts from the developer branch are added into the charts folder of the CATALOG_BRANCH branch.  So the versions of those charts would not be found in the README.md.

```
[ "$(basename $(notdir $@))" = "$$(yq '.name' $(CHART_DIR)/$(basename $(notdir $@))/Chart.yaml)-$$(yq '.version' $(CHART_DIR)/$(basename $(notdir $@))/Chart.yaml)" ]
[ "$$(yq '.version' $(CHART_DIR)/$(basename $(notdir $@))/Chart.yaml)" = "$$(yq '.appVersion' $(CHART_DIR)/$(basename $(notdir $@))/Chart.yaml)" ]
grep "$$(yq '.name' $(CHART_DIR)/$(basename $(notdir $@))/Chart.yaml)[[:space:]]|[[:space:]].*$$(yq '.version' $(CHART_DIR)/$(basename $(notdir $@))/Chart.yaml).*|$$" README.md
```
